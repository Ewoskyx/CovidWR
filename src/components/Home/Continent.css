.continent-main {
  position: relative;
  height: auto;
}

.continent-main:hover {
  box-shadow: rgba(0, 0, 0, 0.25) 0 54px 55px, rgba(0, 0, 0, 0.12) 0 -12px 30px, rgba(0, 0, 0, 0.12) 0 4px 6px, rgba(0, 0, 0, 0.17) 0 12px 13px, rgba(0, 0, 0, 0.09) 0 -3px 5px;
}

.continent-card {
  min-height: 250px;
  background-color: #5788e4;
  box-shadow: rgba(0, 0, 0, 0.17) 0 -23px 25px 0 inset, rgba(0, 0, 0, 0.15) 0 -36px 30px 0 inset, rgba(0, 0, 0, 0.1) 0 -79px 40px 0 inset, rgba(0, 0, 0, 0.06) 0 2px 1px, rgba(0, 0, 0, 0.09) 0 4px 2px, rgba(0, 0, 0, 0.09) 0 8px 4px, rgba(0, 0, 0, 0.09) 0 16px 8px, rgba(0, 0, 0, 0.09) 0 32px 16px;
}

.continent-card p {
  font-family: 'Cabin', sans-serif;
}

.link {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 100;
  border-left: 1px solid #dedede;
  border-top: 1px solid #dedede;
  border-radius: 8px;
  opacity: 0.6;
}
