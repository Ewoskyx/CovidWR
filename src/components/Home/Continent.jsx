import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import './Continent.css';

const Continent = ({
  continent, title, countryCount, totalCountries, totalDeaths, updateDate, id, clickHandler,
}) => (

  <div className="col-12 col-md-5 continent-main">
    <Link to="/countries" onClick={clickHandler} className="link" id={id} />
    <div className="card continent-card text-white">
      <img className="card-img" src={continent} alt={`${title} card`} />
      <div className="card-img-overlay">
        <h5 className="card-title">{title}</h5>
        <p className="card-text">{`Available Countries: ${totalCountries}`}</p>
        <p className="card-text">{`Confirmed Cases: ${countryCount}`}</p>
        <p className="card-text">{`Total Deaths: ${totalDeaths}`}</p>
        <p className="card-text">{`Last Update ${updateDate}`}</p>
      </div>
    </div>
  </div>

);

export default Continent;

Continent.propTypes = {
  continent: PropTypes.string,
  title: PropTypes.string,
  countryCount: PropTypes.number,
  totalCountries: PropTypes.number,
  totalDeaths: PropTypes.number,
  updateDate: PropTypes.string,
  id: PropTypes.string,
  clickHandler: PropTypes.func,
};

Continent.defaultProps = {
  continent: '',
  title: '',
  countryCount: 0,
  totalCountries: 0,
  totalDeaths: 0,
  updateDate: '',
  id: '',
  clickHandler: () => {},
};
