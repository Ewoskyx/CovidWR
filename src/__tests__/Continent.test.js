import '@testing-library/jest-dom/extend-expect';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import Continent from '../components/Home/Continent';
import store from '../redux/configStore';

const renderer = require('react-test-renderer');

it('Renders the Continent component', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <BrowserRouter>
        <Continent />
      </BrowserRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders home component', () => {
  render(<Provider store={store}><BrowserRouter><Continent /></BrowserRouter></Provider>);
  const element = screen.getByText(/Available Countries/i);
  expect(element).toBeInTheDocument();
});
