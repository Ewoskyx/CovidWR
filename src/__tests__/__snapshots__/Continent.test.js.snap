// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Continent component 1`] = `
<div
  className="col-12 col-md-5 continent-main"
>
  <a
    className="link"
    href="/countries"
    id=""
    onClick={[Function]}
  />
  <div
    className="card continent-card text-white"
  >
    <img
      alt=" card"
      className="card-img"
      src=""
    />
    <div
      className="card-img-overlay"
    >
      <h5
        className="card-title"
      >
        
      </h5>
      <p
        className="card-text"
      >
        Available Countries: 0
      </p>
      <p
        className="card-text"
      >
        Confirmed Cases: 
      </p>
      <p
        className="card-text"
      >
        Total Deaths: 
      </p>
      <p
        className="card-text"
      >
        Last Update 
      </p>
    </div>
  </div>
</div>
`;
